filenamify.js

```diff
@@ -37,4 +41,27 @@ const filenamify = (string, options = {}) => {
 	return string;
 };
 
+const recursivelyFilenamify = (string, options = {}) => {
+	const replacement = options.replacement === undefined ? '!' : options.replacement;
+	const substrings = string.split(replacement);
+	const tmp = [];
+	for (let substringIndex = 0; substringIndex < substrings.length; substringIndex++) {
```

Comment:
Use a **for-of** loop.

Reply:
> Use a **for-of** loop.

If it's possible, may I sincerely to request you to set this as a linter rule? This can potentially save much time for both contributors and project maintainers.

Reply:
I already have that. I'm not sure why it didn't detect this one: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-for-loop.md

Reply:
If that's enabled by default, which I believe it is and should be, maybe you want to check if there is a bug in that code. The code here can be used as a test case.

Reply:
> I already have that. I'm not sure why it didn't detect this one: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-for-loop.md

OK, so in the `package.json` of this utility or tool, it depends on `"xo": "^0.24.0"`, which is [released in 14 Jan 2019](https://github.com/xojs/xo/releases/tag/v0.24.0). However, [the **no-for-loop** rule](https://github.com/sindresorhus/eslint-plugin-unicorn/commits/main/rules/no-for-loop.js) you added was on 18 Feb 2019. 

I can see that it kinda pain to manage this many JS packages...

Reply:
Ah right. And I cannot upgrade XO until I drop Node.js 8 support.